Update Linear and Logistic Regression Parameters & Improve Documentation (#8982)
opensearch-project/documentation-website: _ml-commons-plugin/algorithms.md

:--- |:--- | :--- | :---
`target` | String | The name of the target variable to predict. Identifies which feature the model will learn to predict during training. | `NA`
`learning_rate` | Double | The initial step size used in an iterative optimization algorithm. | `0.01`
`momentum_factor` | Double | The extra weight factors that accelerate the rate at which the weight is adjusted. This helps move the minimization routine out of local minima.  | `0`
`epsilon` | Double | The value for stabilizing gradient inversion. | `1.00E-06`
`beta1` | Double | The exponential decay rates for the moment estimates. |  `0.9`
`beta2` | Double | The exponential decay rates for the moment estimates. |  `0.99`
`decay_rate` | Double | The Root Mean Squared Propagation (RMSProp). | `0.9`
`momentum_type` | String | The defined Stochastic Gradient Descent (SGD) momentum type that helps accelerate gradient vectors in the right directions, leading to a fast convergence.| `STANDARD`
`optimiser` | String | The optimizer used in the model. | `SIMPLE_SGD`
`objective` | String | The Objective function used. | `SQUARED_LOSS` 
`epochs` | Integer | The number of iterations. | `5`|

> `batch_size` | Integer | The size of min batches. | `1`

Reviewer: ```suggestion
`batch_size` | Integer | The minimum batch size. | `1`
```

A classification algorithm, logistic regression models the probability of a discrete outcome given an input variable. In ML Commons, these classifications include both binary and multi-class. The most common is the binary classification, which takes two values, such as "true/false" or "yes/no", and predicts the outcome based on the values specified. Alternatively, a multi-class output can categorize different inputs based on type. This makes logistic regression most useful for situations where you are trying to determine how your inputs fit best into a specified category. 

> **Optimisers supported:** `SIMPLE_SGD`, `LINEAR_DECAY_SGD`, `SQRT_DECAY_SGD`, `ADA_GRAD`, `ADA_DELTA`, `ADAM`, and `RMS_PROP`.  

Reviewer: Could we add a description for each optimizer, objective, and momentum type so the user can choose appropriately?

Author: Thanks for the feedback! I think detailed descriptions for each optimizer, objective, and momentum type aren't necessary for the following reasons:

1) External Resources: These concepts are well-documented elsewhere (e.g., Tribuo’s documentation), and adding more here would be redundant. Users can easily find more detailed explanations through a quick search.

2) Experimentation: The best choices often depend on the dataset and require experimentation. Over-explaining in the documentation could make it overly long and less practical.

3) Focus on Functionality: It’s more useful to highlight the available options and let users experiment, rather than dive into detailed descriptions that are readily available online.

Let me know your thoughts!

Reviewer: If this information is available elsewhere, could we link to the descriptions? Ideally, documentation should be self-contained and suitable for experienced as well as novice users. While experienced users know this already, novice users may be turned off by having to search for each of these options' descriptions. Ideally, a one-sentence description should suffice, and for more information users can search elsewhere. Or we can link to a page that contains all these descriptions.

Author: Understood, I will try to either write a one line description or find a link to the documentation.

**Objectives supported:** `HINGE`, and `LOGMULTICLASS`.  
**momentum_type supported:** `STANDARD`, and `NESTEROV`.  

### Parameters

| Parameter | Type | Description | Default value |
|---|---|---|---|
| `learning_rate` | Double | The initial step size used in an iterative optimization algorithm. | `1` |
| `momentum_factor` | Double | The extra weight factors that accelerate the rate at which the weight is adjusted. This helps move the minimization routine out of local minima. | `0` |
| `epsilon` | Double | The value for stabilizing gradient inversion. | `0.1` |
| `beta1` | Double | The exponential decay rates for the moment estimates. | `0.9` |
| `beta2` | Double | The exponential decay rates for the moment estimates. | `0.99` |
| `decay_rate` | Double | The Root Mean Squared Propagation (RMSProp). | `0.9` |

> | `momentum_type` | String | The Stochastic Gradient Descent (SGD) momentum that helps accelerate gradient vectors in the right direction, leading to faster convergence between vectors. | `STANDARD` |

Reviewer: ```suggestion
| `momentum_type` | String | The Stochastic Gradient Descent (SGD) momentum that helps accelerate gradient vectors in the correct direction, leading to faster convergence between vectors. | `STANDARD` |
```

> | `optimiser` | String | The optimizer used in the model.  | `ADA_GRAD` |

Reviewer: Is this the correct parameter name spelling? The American spelling is "optimizer".

Author: Yes, it is "optimiser".

Reviewer 2: @rithin-pullela-aws @kolchfa-aws Please use American English spelling in all instances: "optimizer" in this case. Thanks!

Author: @natebower @kolchfa-aws, since these changes should be back ported to previous versions, I believe it is better to leave it as it might cause backward compatibility issues.  We can make this change in the newer versions.

> | `momentum_type` | String | The Stochastic Gradient Descent (SGD) momentum that helps accelerate gradient vectors in the correct direction, leading to faster convergence between vectors. | `STANDARD` |

Reviewer: ```suggestion
| `momentum_type` | String | The Stochastic Gradient Descent (SGD) momentum that helps accelerate gradient vectors in the right directions, leading to a fast convergence. | `STANDARD` |
```

| `target` | String | The target field. | null |
| `objective` | String | The objective function type. | `LOGMULTICLASS` |
| `epochs` | Integer | The number of iterations. | `5` |

> | `batch_size` | Integer | The size of min batches. | `1` |

Reviewer: ```suggestion
| `batch_size` | Integer | The minimum batch size. | `1` |
```
